symphony/lib/boot/defines.php

```diff
@@ -211,11 +211,32 @@
 define_safe('HTTPS', server_safe('HTTPS'));
 
 /**
- * Returns the current host, ie. google.com
+ * Returns the current host, e.g. google.com
  * @var string
  */
 $http_host = server_safe('HTTP_HOST');
-define_safe('HTTP_HOST', function_exists('idn_to_utf8') ? idn_to_utf8($http_host) : $http_host);
+if (function_exists('idn_to_utf8')) {
```

Comment:
My suggestion is still: `if (extension_loaded('intl')) {`

* if nobody knows a library that can fill the void
* if you have no documentation for it
* if you do not have an unit test for it

Btw. the "Intl" extension can be checked during the installation process.

Reply:
I will be happy to add to the pull request as soon as there is a sort of "common opinion" on the matter. :-)

Reply:
@froschdesign Please do not take my reply the wrong way. I really want to learn and find what is best for the project. @michael-e knows how much I like to argue: but I never do it in a mean/bad way. I want to learn and keep getting better, that is all.

> If nobody knows a library that can fill the void

I feel like user _could_ provide their own version of idn_to_utf8. And this would not be possible.

>  if you have no documentation for it

We do not. Never do we require it. It is not a formal dependancy. Symphony works without it. (That's why is it not checked in the installer, but things like xslt are). Should we require it ? I would not mind to do that for 3.0.x.

> if you do not have an unit test for it

Indeed we do not. But do we really need to test php's code ? Parts of me says yes, part of me says YAGNI.

Finally, since this is LTS, I fear about changing this. There is so much voodoo out there. That's why I would keep it as it is.

Please, continue to weigh in.

Reply:
@nitriques 
> But do we really need to test php's code ?

Oh no, you would not test PHP's code here. You would test the behaviour of the entire code block / workflow. And this with different input and output values (valid and invalid) and also with the user-definied function respectively the existing of the "Intl" extension - which is much easier to test. 😉 

> Finally, since this is LTS...

But LTS does not mean that you need to support dead PHP versions! 😄

Reply:
> But LTS does not mean that you need to support dead PHP versions!

Indeed, but that's not what was agreed upon. Anyhow, 2.7.8 will be the last bug fix release! Thanks for your input @froschdesign